brave/brave-ios
# Fix #1674: Add auto-contribute exclusion list table

This PR also updates the UI in auto-contribute details, bringing the settings up to top-level, and moving supported sites into its own screen

## Summary of Changes

This pull request fixes issue #1674 

Given the test flows required, up to QA whether or not this should go into 1.15 or 1.16 (cc @srirambv  @kjozwiak)

## Submitter Checklist:


## Test Plan:

- Enable rewards, visit some sites to get some auto-contribute publishers, visit a few pre-excluded sites (such as amazon.com)
- Go to Auto-contribute details, verify count on main page matches number of publishers inside supported/excluded list
- Manually exclude all supported publishers. Verify that upon removing the final one, the empty state is shown (edit button also should be disabled)
- Upon backing out to AC details screen, verify publisher counts update
- Enter excluded publisher list, manually restore each. Verify that upon restoring the final one, the empty state is shown.
- Verify that restoring them _does not_ place them in the supported (since the activity of said publisher would be reset to 0)

## Screenshots:

| Base | Empty Supported | Empty Excluded |
| --- | --- | --- |
| ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-20 at 14 25 58](https://user-images.githubusercontent.com/529104/72753006-b3240380-3b91-11ea-9d8e-e569364e98a1.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-20 at 14 26 01](https://user-images.githubusercontent.com/529104/72753009-b61ef400-3b91-11ea-8941-e17f8d542d1f.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-20 at 14 26 04](https://user-images.githubusercontent.com/529104/72753015-b9b27b00-3b91-11ea-9eb0-1ae20a50a27a.png) |

| Supported List | Excluded List | Excluded List Edit |
| --- | --- | --- |
| ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-20 at 14 27 27](https://user-images.githubusercontent.com/529104/72753078-e9fa1980-3b91-11ea-8621-0e63ae99630a.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-20 at 14 27 58](https://user-images.githubusercontent.com/529104/72753081-ec5c7380-3b91-11ea-817e-827c3cf635fb.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-20 at 14 28 05](https://user-images.githubusercontent.com/529104/72753083-eebecd80-3b91-11ea-834f-61dbc91395aa.png) |

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
